### PR TITLE
Fixes #31672 - define ansible roles to ignore

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

File: app/controllers/foreman_ansible/concerns/ansible_roles_controller_extensions.rb [hunk 1]
```diff
@@ -38,11 +38,22 @@ def get_roles_variables(imported_variables, kind, role)
         variables_to_s(variables)
       end
 
+      def excluded_roles
+        @excluded_roles ||= Setting.convert_array_to_regexp(
+          Setting[:ansible_roles_to_ignore],
+          {
+            :prefix => '(\A|.*::|.*_)',
```
Comment (xprazak2) [inline app/controllers/foreman_ansible/concerns/ansible_roles_controller_extensions.rb:45]: Why `::` and `_`? I can't figure that out...

**Comment by xprazak2** [comment]
Works, #382 should go in first. It would be nice to have a test we can correctly exclude roles.

File: app/controllers/foreman_ansible/concerns/ansible_roles_controller_extensions.rb (hunk 1 above)
Comment (shiramax) [reply app/controllers/foreman_ansible/concerns/ansible_roles_controller_extensions.rb]: that's not necessary. I removed it, thanks.

File: app/models/setting/ansible.rb [hunk 2]
```diff
@@ -80,6 +81,12 @@ def default_settings
                'with Ansible inventory'),
             'Ansible - Ansible Inventory',
             N_('Default Ansible inventory report template')
+          ),
+          set(
+            'ansible_roles_to_ignore',
+            N_('Those Roles will be excluded when importing roles from smart proxy, you can use you can use * wildcard metacharacters'),
```
Comment (ares) [inline app/models/setting/ansible.rb:87]: It should probably mention how to specify multiple values. E.g. comma separated values wrapped in quotes or similar. An example wouldn't hurt :-) feel free to use `theforeman.foreman_scap_client` and `theforeman.foreman.*` for that example

**Comment by xprazak2** [comment]
Needs a rebase now.

**Comment by shiramax** [comment]
@xprazak2 done.

**Review by ares — CHANGES_REQUESTED** [review]
It seems to work fine except for one edge case. When I set some roles to be ignored, they are correctly ignored, the wildcards also work fine. But when I reset this to empty value again, all roles seems to be ignored. It seems like `[]` is considered as all. Setting it back to something that doesn't exist (e.g. "a") helps.

The problem is that `Setting.convert_array_to_regexp(Setting[:ansible_roles_to_ignore])` responds with //, which matches everything. The `excluded_roles` should probably have a check for this case. Or we may want to modify the convert_array_to_regexp behavior in core.

**Comment by shiramax** [comment]
@ares thanks for the review, I believe I fixed it now.

**Review by ares — APPROVED** [review]
Nice, it seems tests are now failing due to rubocop issue in REX, but I think they were green before. It may not be a bad idea to add a test for empty string and nil being a setting, but leaving that for another PR. Merging this now, thanks @shiramax!
